hax/treemakers/flash_identification.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import hax
+import pytz
+
+class flash_identification(hax.minitrees.TreeMaker):
+    """
+    Identification of flashes during a dataset. Therefore it will extract trigger data checking for typical
+    flashing properties. It will provide additional information for each event:
+    - flashing_PMT
+    - flashing_time (timestamp)
+    - flashing_width (in seconds)
+    - inside_flash: Events that are within the flash and should be cut in any case
+    - nearest_flash: to set a time window around a flash which can be simply adjusted by a cut (in nseconds)
+    """
+    __version__ = '0.1'
+    pax_version_independent = True
+
+    def get_data(self, dataset, event_list=None):
+        #Use 'all' pulses
+        trigger_data = hax.trigger_data.get_trigger_data(dataset, select_data_types='count_of_all_pulses')
+        self.transposed = trigger_data.T
+
+        #Get BUSY_ON (channel id 255)
+        self.BUSY_data = self.transposed[255]
+
+        time_window = []
+        window_biggest = []
+
+        # Check in BUSY_ON channel if there where an increase "typical" for flashes and get the time-window in seconds
+        for t in range(0,len(self.BUSY_data)):
+            if self.BUSY_data[t] > 20:
```

Comment:
Why 20? Shouldn't a lower number also indicate a problem? Like 5?

Wondering if some smaller flashes won't hit this threshold

Reply:
All flashes I found were above this threshold. It does also not change over time.
If I use a lower threshold I get quite some other issues, which would lead in quite some loss of time if I cut a bigger time-window around all of them. Is this not already accounted for in the DAQ cut?

Reply:
Should be. Was just curious what kinds of things you found. I only see a handful of runs in SR1 with DAQ issues. SR0 is much higher though so probably this threshold makes sense for extending to SR0 too.